pkg/kv/local/store.go

```diff
@@ -0,0 +1,229 @@
+package local
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"time"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+func partitionRange(partitionKey []byte) []byte {
+	return append(partitionKey, kv.PathDelimiter)
+}
+
+func composeKey(partitionKey, key []byte) []byte {
+	return append(partitionRange(partitionKey), key...)
+}
+
+type Store struct {
+	db           *badger.DB
+	logger       logging.Logger
+	prefetchSize int
+	refCount     int
+	path         string
+}
+
+func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWithPredicate, error) {
+	k := composeKey(partitionKey, key)
+	start := time.Now()
+	log := s.logger.WithField("key", string(k)).WithField("op", "get").WithContext(ctx)
+	log.Trace("performing operation")
+	if len(partitionKey) == 0 {
+		log.WithError(kv.ErrMissingPartitionKey).Warn("got empty partition key")
+		return nil, kv.ErrMissingPartitionKey
+	}
+	if len(key) == 0 {
+		log.WithError(kv.ErrMissingKey).Warn("got empty key")
+		return nil, kv.ErrMissingKey
+	}
+
+	var value []byte
+	err := s.db.View(func(txn *badger.Txn) error {
+		item, err := txn.Get(composeKey(partitionKey, key))
+		if errors.Is(err, badger.ErrKeyNotFound) {
+			return kv.ErrNotFound
+		}
+		if err != nil {
+			log.WithError(err).Error("error getting key")
+			return err
+		}
+		value, err = item.ValueCopy(nil)
+		if err != nil {
+			log.WithError(err).Error("error getting value for key")
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		log.WithField("took", time.Since(start)).Trace("operation failed")
+		return nil, err
+	}
```

Comment:
Maybe it is the extensive logging - but this one for example prints "operations failed" for key not found, which is a bit misleading.